index.js

```diff
@@ -26,24 +26,33 @@ EventedRequire.prototype.require = function(moduleName) {
     opts.basedir = this._basedir;
   }
 
-  var result;
+  var result, already, modulePath;
   this.emit('before', moduleName);
   try {
-    result = require(resolve.sync(moduleName, opts));
+    modulePath = resolve.sync(moduleName, opts);
+    if (require.cache[modulePath]) {
+      already = true;
+    }
+    result = require(modulePath);
   } catch (e) {
     this.emit('failure', moduleName, e);
-    return;
+    return null;
+  }
+
+  if (already) {
+    this.emit('already', moduleName, modulePath);
```

Comment:
I'm trying to think of a better name for this.  Maybe `exists` - I'm open to suggestions.

Reply:
I changed the event name to `exists`.